corehq/motech/repeaters/models.py

```diff
@@ -477,14 +478,32 @@ def handle_response(self, result, repeat_record):
 
         result may be either a response object or an exception
         """
+        _4XX_retry_codes = (
+            HTTPStatus.REQUEST_TIMEOUT,
+            HTTPStatus.CONFLICT,
+            HTTPStatus.PRECONDITION_FAILED,
+            HTTPStatus.LOCKED,
+            HTTPStatus.FAILED_DEPENDENCY,
+            HTTPStatus.TOO_EARLY,
+            HTTPStatus.UPGRADE_REQUIRED,
+            HTTPStatus.PRECONDITION_REQUIRED,
+            HTTPStatus.TOO_MANY_REQUESTS,
+        )
+
         if isinstance(result, RequestConnectionError):
             repeat_record.handle_timeout(result)
         elif isinstance(result, Exception):
             repeat_record.handle_exception(result)
-        elif is_response(result) and 200 <= result.status_code < 300 or result is True:
+        elif is_success_response(result):
             repeat_record.handle_success(result)
-        else:
+        elif not is_response(result) or (
+            500 <= result.status_code < 600
+            or result.status_code in _4XX_retry_codes
+        ):
             repeat_record.handle_failure(result)
+        else:
+            message = format_response(result)
+            repeat_record.handle_payload_error(message)
```

Comment:
So this is basically saying if this is something like a 401 or 404, it is a payload error and we should just cancel that record immediately right?

Reply:
Yes exactly. We retry if there is a 5XX error or a 4XX error that might succeed if we try again later, but if the problem is the payload, retrying isn't going to help.